File: backend/src/main/java/com/service/NotificationService.java
package com.service;

import com.model.Notification;
import java.util.List;

/**
 * Service defining notification operations for recipient inboxes.
 */
public interface NotificationService {

    void notify(String recipientId, String senderName, String type, String message, String referenceId);

    List<Notification> getNotifications(String email);

    void markAllAsRead(String email);

    void markAsRead(String email, String notificationId);

    void deleteNotification(String email, String notificationId);

    void deleteAllNotifications(String email);
}

File: frontend/src/services/notificationService.js
import api from './api';

/**
 * Service to retrieve and clear user notification center logs.
 */

export const getNotifications = () => {
  return api.get('/api/notifications');
};

export const markAllAsRead = () => {
  return api.post('/api/notifications/read');
};

export const markAsRead = (id) => {
  return api.post(`/api/notifications/${id}/read`);
};

export const deleteNotification = (id) => {
  return api.delete(`/api/notifications/${id}`);
};

export const deleteAllNotifications = () => {
  return api.delete('/api/notifications');
};
